### PR TITLE
Add interactive standalone keystone mode

### PR DESCRIPTION
**Pull request by ErikMcClure**
This still does not allow installing keystone as a service, so it must be run under a session. You can do this via `cargo run` via the following:

```
cargo run --features="binary-deps" --bin keystone -- session -i -t .\examples\example.windows.toml
```

the actual command here is:

```
keystone session -i -t .\examples\example.windows.toml
```

Keystone will still run without `-i` but it won't be interactive. `-i` provides the interactive TUI that lets you manage the modules you spawned.

Keystone is still using stdin/stdout for RPC communication, which will be addressed in a separate PR. This PR contains necessary refactors to make it possible to fix that.

### CODE REVIEW

File: core/src/main.rs [hunk 1]
```diff
@@ -157,18 +175,1130 @@ fn inspect<R: Read>(
...
+                        if app.instance.modules.contains_key(id) {
+                            None
+                        } else {
+                            Some(*id)
+                        }
+                    })
+                    .collect::<Vec<_>>();
+
+                if let Ok((id, line)) = log_rx.try_recv() {
+                    if let Some(m) = app.modules.get_mut(&id) {
```
Comment (LunNova) [inline core/src/main.rs:1132]: Is it ok we can receive a log line and then have it disappear if this check fails?

File: core/src/main.rs [hunk 2]
```diff
@@ -157,18 +175,1130 @@ fn inspect<R: Read>(
...
+    }
+
+    Ok(())
+}
+
+fn log_capture(id: u64, log_tx: UnboundedSender<(u64, String)>) -> impl LogCapture {
+    move |mut input: ByteStreamBufferImpl| {
+        let send = log_tx.clone();
+        async move {
+            let mut buf = [0u8; 1024];
```
Comment (LunNova) [inline core/src/main.rs:1224]: What happens if a line is longer than 1024? Fixed size buf, only writes out once hitting a new line :thinking:

File: core/src/main.rs [hunk 3]
```diff
@@ -157,18 +175,1130 @@ fn inspect<R: Read>(
...
+                        if let Ok(s) = std::str::from_utf8(&buf[..n]) {
+                            line += s;
+                        }
+                        buf.copy_within(n..len, 0);
+                        break;
+                    } else if let Ok(s) = std::str::from_utf8(&buf[..len]) {
+                        line += s;
+                    }
+                }
+                // This IS the log function so we can't really do anything if this errors
```
Comment (LunNova) [inline core/src/main.rs:1245]: eprintln!?

File: core/src/module.rs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            let _ = p.kill_request().send().promise.await;
+        }
+
+        self.state = ModuleState::Aborted;
+    }
+
+    pub async fn pause(&mut self, pause: bool) -> Result<()> {
+        self.pause.send(pause).await?;
+
+        if pause {
```
Comment (LunNova) [inline core/src/module.rs:168]: `pause(false)` as API to unpause seems a bit odd.

this doesn't check the current state so you can go from Aborted -> pause(false) -> ready which seems like an incorrect transition to make

nit i prefer `self.state = if pause { ModuleState::Paused } else { ModuleState::Ready };`, unimportant

File: core/src/main.rs (hunk 2 above)
Comment (ErikMcClure) [reply core/src/main.rs:1224]: It is appended to a string called line, and the string is written out once it hits a newline. I double checked this by making the buffer size 16 characters and it worked fine.

File: core/src/main.rs (hunk 3 above)
Comment (ErikMcClure) [reply core/src/main.rs:1245]: You can't eprintln in an interactive console TUI, there is nowhere for it to go. This code is only used when the interactive console is active.

File: core/src/main.rs (hunk 1 above)
Comment (ErikMcClure) [reply core/src/main.rs:1132]: There is nowhere to store the log if it was emitted for a module that doesn't exist. Again, the interactive UI means we cannot simply dump to stderr.

**Review by LunNova — APPROVED** [review]

**Pull request merged**